OpenRA.Mods.Common/Traits/BotModules/Squads/States/ProtectionStates.cs

```diff
@@ -56,9 +92,56 @@ public void Tick(Squad owner)
 			}
 			else
 			{
+				cannotRetaliate = true;
+
 				foreach (var a in owner.Units)
-					owner.Bot.QueueOrder(new Order("AttackMove", a, Target.FromCell(owner.World, owner.TargetActor.Location), false));
+				{
+					// Air units control:
+					if (a.Info.HasTraitInfo<AircraftInfo>() && a.TraitsImplementing<AmmoPool>().Any())
+					{
+						if (BusyAttack(a))
+						{
+							cannotRetaliate = false;
+							continue;
+						}
+
+						if (!ReloadsAutomatically(a))
```

Comment:
I would pull out a `var ammoPools = a.TraitsImplementing<AmmoPool>().ToArray();` here before this `if` and pass it to the boolean methods as argument, so they don't have to each look up the ammo pools themselves.
This avoids enumerating the AmmoPools up to twice (once in `ReloadsAutomatically` and once in `HasAmmo`).